tests/postUI.test.tsx

```diff
@@ -0,0 +1,145 @@
+    expect(screen.getByLabelText('request invitations')).toBeDefined();
+  });
+
+  it("Renders with only time until event starts", () => {
+    render(<PostItem item={post} likes={0} reportPost={mockReportPost}
+      writtenByYou={false} operations={mockOps} startTime={time}
+      maxOccupancy={undefined} hasInvitations={false}
+    />);
+    
+    expect(screen.getByText(props.description)).toBeDefined();
+    expect(screen.queryByLabelText('time until')).toBeDefined();
+    expect(screen.queryByLabelText('max occupancy')).toBeNull();
+    expect(screen.queryByLabelText('request invitations')).toBeNull();
+  });
+
+  it("Renders with only max occupancy", () => {
+    render(<PostItem item={post} likes={0} reportPost={mockReportPost}
+      writtenByYou={false} operations={mockOps} startTime={undefined}
+      maxOccupancy={8} hasInvitations={false}
+    />);
+    
+    expect(screen.getByText(props.description)).toBeDefined();
+    expect(screen.queryByLabelText('time until')).toBeNull();
+    expect(screen.queryByLabelText('max occupancy')).toBeDefined();
+    expect(screen.queryByLabelText('request invitations')).toBeNull();
+  });
+
+  it("Renders with only invitations", () => {
+    render(<PostItem item={post} likes={0} reportPost={mockReportPost}
+      writtenByYou={false} operations={mockOps} startTime={undefined}
+      maxOccupancy={undefined} hasInvitations={true}
+    />);
+    
+    expect(screen.getByText(props.description)).toBeDefined();
+    expect(screen.queryByLabelText('time until')).toBeNull();
+    expect(screen.queryByLabelText('max occupancy')).toBeNull();
+    expect(screen.queryByLabelText('request invitations')).toBeDefined();
+  });
+
+  it("Name is shortened", () => {
+    render(<PostItem item={post} likes={0} reportPost={mockReportPost}
+      writtenByYou={false} operations={mockOps} startTime={time}
+      maxOccupancy={8} hasInvitations={true}
+    />);
+    
+    expect(screen.getByText("Post T.")).toBeDefined();
+  });
+
+  it("Color of invitation changes on click", () => {
+    render(<PostItem item={post} likes={0} reportPost={mockReportPost}
+      writtenByYou={false} operations={mockOps} startTime={undefined}
+      maxOccupancy={undefined} hasInvitations={true}
+    />);
+    const invitation = screen.queryByLabelText('invitation icon');
+
+    expect(invitation.props.style[0].color).toEqual("black");
+    fireEvent.press(invitation);
+    expect(invitation.props.style[0].color).toEqual(generateColor);
+  });
+
+  it("Time becomes red", () => {
+    time.setMinutes(time.getMinutes() + 30);
+    render(<PostItem item={post} likes={0} reportPost={mockReportPost}
+      writtenByYou={false} operations={mockOps} startTime={time}
+      maxOccupancy={undefined} hasInvitations={false}
+    />);
+    const timeIcon = screen.queryByLabelText('time icon');
+
+    expect(timeIcon.props.style[0].color).toEqual("red");
+  });
+
+  it("Occupancy becomes red", () => {
```

Comment:
It would be nice if we could somehow map each of these test cases to a storybook story, which should make testing easier. What do you think @mhayes853

Reply:
That’s a very good point distinguishing between styling and functionality changes, and I agree with it. I was more intending to call out UI edge cases that would be covered by both the storybook and the unit tests. Ex. If we pass in a very long name will it be truncated, will long descriptions be truncated, will the “invite” button be hidden if the event is public, error/loading states, etc. we’d typically add those cases to the storybook but I could also see benefits to turning those into automated tests.

Reply:
Those examples would be considered functionality. What I mean by styles in particular are minute details like the position of an element, or the grid (padding) between elements, etc. In essence, we don't need to test what every piece of text is for a particular component, only the important pieces based on the functionality.